website/client-old/js/controllers/challengesCtrl.js

```diff
@@ -493,7 +493,7 @@ habitrpg.controller("ChallengesCtrl", ['$rootScope','$scope', 'Shared', 'User',
           .then(function (response) {
             var challenge = response.data.data;
             $scope.challenges = [challenge];
-          });
+          }).catch(function() { $scope.challengeNotFound = true; });
```

Comment:
The catch function should pass an error. Would it be better to pass on the error message from the API call? That would provide better reporting if the request failed for another reason.